server/clientRoutes.js

```diff
@@ -117,15 +128,78 @@ function clientsRouter(clients, defaultId) {
   });
 
   // get info about a specific client
-  router.get('/:id', (req, res) => {
+  router.get('/:id', async (req, res) => {
+    let configurations;
+    try {
+      configurations = await getConfig(req.params.id);
+    } catch (e) {
+      logger.error(e);
+      if (e.code === 'ENOENT')
+        return res.status(404).send(`Config for "${req.params.id}" not found`);
+      else
+        return res.status(500).send(`There was a problem with your request.`);
+    }
+
     const info = {
-      network: config.str('network'),
-      chain: config.str('chain', 'bitcoin')
+      configs: configurations.data
     };
+
+    if (req.query.checkStatus) {
+      logger.info(`Checking status of client "${req.params.id}"...`);
+      try {
+        await testConfigOptions(configurations);
+        info.status = true;
+      } catch (e) {
+        if (e.failed && e.failed.length) {
+          info.failed = e.failed;
+          info.status = false;
+        } else {
+          return res.status(500).send(e);
+        }
+      }
+    }
+
+    // scrub apiKeys and tokens
```

Comment:
I don't think we should use the `delete` keyword, V8 experiences overhead when you use it and in my experience its something to be avoided. Maybe we can set the value to `undefined` instead?

Reply:
yep, works for me. Any preferences between `undefined` or  just empty string `''`?

Reply:
Updated to be `undefined`.

Reply:
No preference between the two, either should work fine, I think that the `json` parser will skip it if its undefined compared to keeping it as an empty string otherwise

Reply:
Yep! It did. Good call